core-bundle/src/DependencyInjection/Compiler/CrawlerPass.php

```diff
@@ -16,20 +16,28 @@
 use Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 
-class EscargotSubscriberPass implements CompilerPassInterface
+class CrawlerPass implements CompilerPassInterface
 {
     use PriorityTaggedServiceTrait;
 
     public function process(ContainerBuilder $container): void
     {
-        if (!$container->has('contao.crawl.escargot_factory')) {
+        $subscribers = $this->findAndSortTaggedServices('contao.escargot_subscriber', $container);
+
+        if (!$container->has('contao.crawl.escargot_factory') || 0 === \count($subscribers)) {
+
+            // Remove factory
+            $container->removeDefinition('contao.crawl.escargot_factory');
```

Comment:
Not sure if this works. Maybe the has check needs to be extra but tests will show.

Reply:
I am a bit confused. If the container does **not** have the definition, remove the definition?